src/FHIRDefinitions.ts

```diff
@@ -392,9 +392,14 @@ export class FHIRDefinitions {
     const defsToSearch: FHIRDefinitions[] = [this];
     while (defsToSearch.length > 0) {
       const currentFHIRDefs = defsToSearch.shift();
-      const def = currentFHIRDefs[map].get(item);
+      const [base, ...versionParts] = item?.split('|') ?? ['', ''];
+      const version = versionParts.join('|') || null;
+      const def = currentFHIRDefs[map].get(base);
       if (def) {
-        return def;
+        if (version == null || def.version == null || version === def.version) {
```

Comment:
I was actually thinking about this again yesterday/today. I originally did it because (a) it was how SUSHI was already checking if it should log a warning message if a different version was fished up and (b) I was thinking if there was no version specified it could maybe be considered a match because it isn't strictly a _mismatch_. But yesterday and this morning I was starting to wonder if that was too lenient and it should be a _strict match_ rather than a _strict mismatch_, if that makes sense.

So, I'm open to changing it. I don't know exactly if this will be so strict that is is harmful. I can try to use this approach in SUSHI and run a regression and see if issues are caused.

Reply:
I think stricter might be better (i.e., only return matches w/ 100% confidence). But... I think it's a good idea to try it out in SUSHI and see what the regression tells us.

Reply:
The regression had no changes from the previous approach, so I think it is fine to update this. Also, thinking about it a little more, SUSHI still uses the "best version if not an exact match" approach, so I think updating to make this stricter won't change much in SUSHI's implementation and makes sense for FPL's implementation. So I'll push up the change and a new test.

Reply:
Updated in [6f1c5e9](https://github.com/standardhealth/fhir-package-loader/pull/11/commits/6f1c5e9d95659d925207c56a88e94cb15cdac94c).